### PR TITLE
brave_stats: Send Processor Arch if ads enabled

### PR DESCRIPTION
**Pull request by keur**
Resolves https://github.com/brave/brave-browser/issues/14232

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [x] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/brave_stats/brave_stats_updater.cc [hunk 1]
```diff
@@ -265,6 +271,11 @@ bool BraveStatsUpdater::IsReferralInitialized() {
 #endif
 }
 
+bool BraveStatsUpdater::IsAdsEnabled() {
+  return ProfileManager::GetPrimaryUserProfile()->GetPrefs()->GetBoolean(
```
Comment (tmancey) [inline browser/brave_stats/brave_stats_updater.cc:275]: Users could have ads enabled on another profile but disabled on their primary profile. Is this intentional?

File: browser/brave_stats/brave_stats_updater_unittest.cc [hunk 2]
```diff
@@ -72,7 +72,7 @@ TEST_F(BraveStatsUpdaterTest, IsDailyUpdateNeededLastCheckedYesterday) {
   GetLocalState()->SetString(kLastCheckYMD, kYesterday);
 
   brave_stats::BraveStatsUpdaterParams brave_stats_updater_params(
-      GetLocalState(), GetProfilePrefs(), kToday, kThisWeek, kThisMonth);
+      GetLocalState(), GetProfilePrefs(), false, kToday, kThisWeek, kThisMonth);
```
Comment (tmancey) [inline browser/brave_stats/brave_stats_updater_unittest.cc:75]: nit: `/*is_uncertain_future*/ false` (see below too)

File: browser/brave_stats/brave_stats_updater_unittest.cc [hunk 3]
```diff
@@ -199,7 +211,7 @@ TEST_F(BraveStatsUpdaterTest, HasDateOfInstallationFirstRun) {
   SetCurrentTimeForTest(current_time);
 
   brave_stats::BraveStatsUpdaterParams brave_stats_updater_params(
-      GetLocalState(), GetProfilePrefs(), kToday, kThisWeek, kThisMonth);
+      GetLocalState(), GetProfilePrefs(), false, kToday, kThisWeek, kThisMonth);
   ASSERT_EQ(brave_stats_updater_params.GetDateOfInstallationParam(),
```
Comment (tmancey) [inline browser/brave_stats/brave_stats_updater_unittest.cc:215]: nit: `EXPECT_EQ` (See others too)

File: browser/brave_stats/brave_stats_updater.cc [hunk 4]
```diff
@@ -275,23 +286,56 @@ void BraveStatsUpdater::DisableThresholdPing() {
...
-  } else {
+  const bool referrals_initialized = IsReferralInitialized();
+  const bool ads_enabled = IsAdsEnabled();
+  int num_closures = 0;
+
+  // Note: We don't have the callbacks here because otherwise there is a race
+  // condition whereby the callback completes before the barrier has been
+  // initialized.
+  if (!referrals_initialized) {
+    num_closures += 1;
```
Comment (emerick) [inline browser/brave_stats/brave_stats_updater.cc:297]: Nit: Can just use `++num_closures` here and below.

**Review by emerick — APPROVED** [review]
LGTM, just mentioned a nit.

File: browser/brave_stats/brave_stats_updater.cc (hunk 1 above)
Comment (keur) [reply browser/brave_stats/brave_stats_updater.cc:274]: Yeah for now it's fine. It's how we calculated it for the `adsEnabled` param which has given us good results. Stats is also just a heuristic for these things.

**Review by tmancey — APPROVED** [review]
LGTM

**Pull request merged**